src/components/DocExplorer.js

```diff
@@ -43,23 +43,31 @@ export class DocExplorer extends React.Component {
   }
 
   // Public API
-  showDoc(typeOrField) {
-    var navStack = this.state.navStack;
-    var isCurrentlyShown =
-      navStack.length > 0 && navStack[navStack.length - 1] === typeOrField;
-    if (!isCurrentlyShown) {
-      navStack = navStack.concat([ typeOrField ]);
+  showDoc(navItem) {
+    // Creating a deep copy in order to trigger state change when lastEntry is
+    // modified.
+    var navStack = this.state.navStack.slice(0);
+    let lastEntry = navStack.length > 0 && navStack[navStack.length - 1];
+    const isCurrentlyShown = lastEntry === navItem;
+
+    // First check if a navItem argument and the last entry from navStack are
+    // both Search Results - if so, push back the last entry with searchValue
+    // updated.
+    if (navItem.name === 'Search Results' &&
```

Comment:
For example, this seems a bit suspect. This creates a coupling to this particular string. Also, as search expands in features, for example to search within a particular type, it's unlikely that we'll want to always have this title.